core/build.gradle.kts

```diff
@@ -12,6 +12,12 @@ android {
         testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
         consumerProguardFiles("consumer-rules.pro")
         buildConfigField("String", "OTEL_ANDROID_VERSION", "\"$version\"")
+
+// Enable desugaring for Java 8 features
+        compileOptions {
+            sourceCompatibility = JavaVersion.VERSION_1_8 // Set source compatibility to Java 8
+            targetCompatibility = JavaVersion.VERSION_1_8 // Set target compatibility to Java 8
+        }
```

Comment:
Probably not needed because depending on the AGP version, 1.8 is already the default, unless its already Java 11 and you want to force Java 1.8

Reply:
@marandaneto How does the new changes look? I'm still making some improvements, just want to make sure I'm on the right track and following everything right👍